zap/src/lib.rs

```diff
@@ -49,14 +35,12 @@ pub struct Code {
 }
 
 #[cfg(not(target_arch = "wasm32"))]
-pub fn run(config: &str) -> Result<Code, Error> {
-	let file = parser::parse(config)?;
+pub fn run(input: &str) -> (Option<Code>, Vec<Diagnostic<()>>) {
```

Comment:
`(Option<Code>, Vec<Diagnostic<()>>)` should be put into a type.

Reply:
moved into `Return` struct